packages/core/src/browser/label-provider.ts

```diff
@@ -256,6 +270,16 @@ export class DefaultUriLabelProviderContribution implements LabelProviderContrib
     }
 }
 
+/**
+ * The {@link LabelProvider} determines how elemens/nodes are displayed in the workbench. For any element, it can determine a short label, a long label
+ * and an icon. The {@link LabelProvider} is to be used in lists, trees and tables, but also view specific locations like headers.
+ * The common {@link LabelProvider} can be extended/adapted via {@link LabelProviderContribution}s. For every element, the {@links LabelProvider} will determine the
+ * {@link LabelProviderContribution} with the hightest priority and delegate to it. Theia registers default {@link LabelProviderContribution} for common types, e.g.
+ * the {@link DefaultUriLabelProviderContribution} for elements that have a URI.
+ * Using the {@link LabelProvider} across the workbench ensures a common look and feel for elements across multiple views. To adapt the way how specific
+ * elements/nodes are rendered, use a {@link LabelProviderContribution} rather than adapting or sub classing the {@link LabelProvider}. This way, your adaptation
+ * is applied to all view in Theia that use the {@link LabelProvider}
+ */
```

Comment:
This is only one implementation of the `LabelProviderContribution`, I'm not sure it itself should be documented?
The interface would be the common documentation for all contributions.

Reply:
This is a central part of our API, something users might have to actually use (vs implement for the contribution interface). I think it makes sense to add documentation at this place since it will appear when people hover over it, giving good usage info.

Reply:
LabelProvider is not an implementation of LabelProviderContribution. It is the central service to consume LabelProviderContributions. It is also not an interface.